src/boot/efi/pe.c

```diff
@@ -151,9 +151,17 @@ static void locate_sections(
         assert(offsets);
         assert(sizes);
 
+        size_t prev_section_addr = 0;
+
         for (UINTN i = 0; i < n_table; i++) {
                 const PeSectionHeader *sect = section_table + i;
 
+                if (in_memory) {
+                        if (prev_section_addr > sect->VirtualAddress)
```

Comment:
hmm, are sections supposed to be ordered by offset in the file? Id don't think so, do they?

Reply:
As far as I can tell, sections are sorted by VMA.

Reply:
indeed, spec says so. 

https://learn.microsoft.com/en-us/windows/win32/debug/pe-format#section-table-section-headers

Maybe add a comment here, saying sections must be ordered by VirtualAddress, and maybe even add a reference to the spec.

Hmm, maybe also check this order while we iterate through the thing, given such a check is basically free.

Reply:
And if you looked at the code you'd see that it does exactly that. :facepalm: